lib/rubocop/cop/rspec/stubbed_mock.rb

```diff
@@ -157,18 +162,7 @@ def on_expectation(expectation, method_name, matcher)
         def msg(method_name)
           format(MSG,
                  method_name: method_name,
-                 replacement: replacement(method_name))
-        end
-
-        def replacement(method_name)
-          case method_name
```

Comment:
I would prefer a `case`.
Why not?

Reply:
SimpleCov pointed out that we don’t have any test coverage for when `method_name` didn’t match any of the options. And correctly so.

I also prefer the `case` statement, but perhaps we should add an `else raise ArgumentError` or `KeyError` or similar, and add test coverage (where relevant).

Reply:
Ha! We can actually [configure expectations](https://github.com/rubocop/rubocop-rspec/blob/5d96600242e849526a2aaa6e35ecf16c2982ff87/lib/rubocop/rspec/language.rb#L129) DSL, and add eg the mentioned `are_expected`, for which this cop won’t find a substitute.

But there is no substitute for eg are_expected. Can we skip correction in this case instead of throwing an argument error?

`fetch` would raise, and this is also an undesirable outcome, and a breaking change.

I was never a big fan of 100% coverage, but it really starts to manifest its benefits here!

Reply:
Correcting myself. Dsl conf [link](https://github.com/rubocop/rubocop-rspec/blob/5d96600242e849526a2aaa6e35ecf16c2982ff87/config/default.yml#L62). are_expected, and should_* are included.

Quick suggestion: keep the `case`, but limit matching to just the three options we have substitutes for. Wdyt?

Reply:
> Quick suggestion: keep the `case`, but limit matching to just the three options we have substitutes for. Wdyt?

That is what we had before, no? Should we add an `else` case, and how should it respond?

Reply:
Previously I think it was returning nil, and we auto-corrected to nonsense. Now, I think, it should just raise an offence and don’t autocorrect

Reply:
> Now, I think, it should just raise an offence and don’t autocorrect

I'm not exactly sure how to accomplish this. 

Would you like me to close this PR so you can take it a new direction?

Or I'm happy to continue helping out here if you are willing to coach me through it.

Reply:
My bad - this cop does not autocorrect 🙈

Since there’s no valid suggestion for the message:
1 add a new MSG_NO_RePLACEMENT with a different message
2 use it for anything that doesn’t have a built-in counterpart

I insist on keeping the case statement, as ‘fetch’ may conceal execution paths from our coverage detection